lib/common.js

```diff
@@ -203,23 +204,30 @@ async function notify_hooks(pg_pool, app_uuid, type, payload, username) {
         options.headers['x-appkit-signature'] = ('sha1=' + hmac.update(payload).digest('hex'));
 
         // fire the hook, record its result.
+        let recorded_result = false;
         let req = service.request(options, (res) => {
-          let data = new Buffer(0);
+          let data = Buffer.alloc(0)
           res.on('data', (chunk) => Buffer.concat([data, chunk]));
           res.on('end', () => {
-            insert_hook_result(pg_pool, [id, hook.hook, hook.events, hook.url, res.statusCode, JSON.stringify(res.headers), data.toString('utf8'), JSON.stringify(options.headers), payload])
-              .catch((err) => {
-                console.warn('Error recording hook result:', err);
-                console.warn('Data not recorded: ', [id, hook.hook, hook.events, hook.url, res.statusCode, JSON.stringify(res.headers), data.toString('utf8'), JSON.stringify(options.headers), payload]);
-              });
+            if (!recorded_result) {
```

Comment:
It's possible for both the error event and the end event to both fire, this just prevents end from recording the result if an error reporting the hooks happens.